csm_web/scheduler/models.py

```diff
@@ -66,6 +72,22 @@ class Meta:
         abstract = True
 
 
+class Semester(ValidatingModel):
```

Comment:
I'd also add a field for the user-readable name of the semester, since currently it's only defined by the `sid`, which is not related to the meaning/interpretation of the semester object. This could just be a small `CharField`, or something similar; I'm envisioning it to just hold something like `sp22` or `fa20`, etc. You could even add some validation to ensure that it's of the form `(sp|fa)\d\d` if you'd like.